certbot/main.py

```diff
@@ -413,6 +414,9 @@ def _find_domains_or_certname(config, installer, question=None):
     # if we can't do that but we have a certname, get the domains
     # with that certname
     elif certname:
+        if bool(re.search(r'[\\/]', certname)) or bool(re.search('^-', certname)):
```

Comment:
No need for `re` here! Python has the `in` operator, and for the start one a `certname[0]` will do just fine. More pythonic and readable that way. Plus that'll probably fix much of the tests -- escaping and `r''` and all are complicated.

Reply:
`certname[0] == '-'` is a good idiom; another fun option is `certname.startswith('-')`.